sensors/utils/utils.go

```diff
@@ -0,0 +1,51 @@
+// Package utils contains helper functions for the sensor implementations.
+package utils
+
+import (
+	"context"
+	"image"
+
+	"github.com/edaniels/gostream"
+	"github.com/pkg/errors"
+	"go.viam.com/rdk/components/camera"
+	"go.viam.com/rdk/rimage"
+	"go.viam.com/rdk/utils"
+)
+
+// GetName returns the name of the sensor based on its index in the sensor array.
+func GetName(sensors []string, index int) (string, error) {
+	if index > len(sensors) {
+		return "", errors.New("index out of bounds")
+	}
+	return sensors[index], nil
+}
+
+// GetPNGImage first attempts to get a lazy PNG image. If the image is not a lazy PNG, the
+// function expects it to be an YCBCR image. If it is neither a lazy PNG nor an YCBCR, the
+// function errors out. The returned function is a release function that must be called once
+// the caller of GetPNGImage is done using the image.
+func GetPNGImage(ctx context.Context, cam camera.Camera) ([]byte, func(), error) {
```

Comment:
[note to self] This code came from [here](https://github.com/viamrobotics/rdk/blob/b4a5473abe9a4721ddfc5300c87927e52c7511fb/services/slam/builtin/builtin.go#L780)